### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Richa-rani02**
functionality Added: signin and signup with Api integration
User cannot go to wishlist without signup
 
Few commented code and console.log is for doubt clearing purpose pls ignore them

### CODE REVIEW

File: src/utils/getDataFromServer.js [hunk 1]
```diff
@@ -22,4 +22,45 @@ export const getProductFromServer=async(dispatch)=>{
...
+
+export const signupOnServer=async(userDetails,authDispatch)=>{
+    try{
+        console.log(userDetails);
+        const response=await axios.post(signupUrl,
+            {
+              firstName:userDetails.firstName,
+              lastName:userDetails.lastName,
+              email:userDetails.email,
+              password:userDetails.password
```
Comment (AmishMishra11) [inline src/utils/getDataFromServer.js:35]: you could use destructuring so as to avoid typing `userDetails.` in every line

**Review by AmishMishra11 — APPROVED** [review]
Overall very clean implementation.
Liked the way you have done routing for the signup page and good implementation of authentication.
Just remove comments and console before publishing.

**Review by tanishk15sharma — APPROVED** [review]

File: src/context/authContext.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
+import { createContext, useContext, useEffect, useReducer } from "react";
+import { authReducer } from "../reducers/authReducer";
+
+const AuthContext=createContext({});
+const AuthProvider=({children})=>{
+const getoken=localStorage.getItem('sessiontoken')
+    const authInitialState={
+        user:[],
```
Comment (SukanyaSen530) [inline src/context/authContext.js:8]: Why are you using array here? Correct me if I am wrong, I think it should be an object (if you are planning to store the user after login).

Comment (Richa-rani02) [reply src/context/authContext.js:8]: It was trying something forgot to remove this .removing it

**Pull request merged**